basictracer/scope.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2017 The OpenTracing Authors.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+from opentracing import Scope
+
+
+class BasicScope(Scope):
+    """BasicScope is the implementation of `opentracing.Scope`"""
+
+    def __init__(self, manager, span, finish_on_close=True):
+        """Initialize a `Scope` for the given `Span` object
+
+        :param span: the `Span` used for this `Scope`
+        :param finish_on_close: whether span should automatically be
+            finished when `Scope#close()` is called
+        """
+        self._manager = manager
+        self._span = span
+        self._finish_on_close = finish_on_close
+        self._to_restore = manager.active()
+
+    def span(self):
+        """Return the `Span` that's been scoped by this `Scope`."""
+        return self._span
+
+    def close(self):
```

Comment:
Hey - while porting our python-examples repository, I went through this code and wondered if there's a reason to have `Scope.close()` call `ScopeManager.deactivate()` (or similar) instead? That way we would reuse `BasicScope` lots more, I think - by putting specific stuff in `ScopeManager`.

That being said, I could prepare a PR if/as needed on this ;)

Reply:
I.e. specially since it would be nice (*if possible*), to separate `BasicScope` from the actual `ThreadLocalScopeManager` implementation ;)